app/components/post_textbox/post_textbox.js

```diff
@@ -443,11 +451,10 @@ export default class PostTextbox extends PureComponent {
     };
 
     onShowFileMaxWarning = () => {
-        this.setState({showFileMaxWarning: true}, () => {
-            setTimeout(() => {
-                this.setState({showFileMaxWarning: false});
-            }, 3000);
-        });
+        EventEmitter.emit('fileMaxWarning', true);
+        setTimeout(() => {
+            EventEmitter.emit('fileMaxWarning', false);
```

Comment:
Perhaps it would make sense to have the `FileUploadPreview` handle hiding the warning itself? You could remove the parameter and have it do

```javascript
handleFileMaxWarning = () => {
    this.setState({showFileMaxWarning: true});

    setTimeout(() => {
        this.setState({showFileMaxWarning: false});
    }, 3000);
};
```